borsh-derive-internal/src/enum_ser.rs

```diff
@@ -169,9 +179,6 @@ fn unnamed_fields(
         }
     }
     variant_header = quote! { ( #variant_header )};
-    let variant_idx_body = quote!(
-        #enum_ident::#variant_ident(..) => #discriminant_value,
-    );
     Ok(VariantParts {
         variant_header,
```

Comment:
add following 2 tests to `tests` submodule in this file:
```rust
    #[test]
    fn borsh_discriminant_false() {
        let item_enum: ItemEnum = syn::parse2(quote! {
           #[borsh(use_discriminant = false)]
            enum X {
                A,
                B = 20,
                C,
                D,
                E = 10,
                F,
            }
        })
        .unwrap();
        let actual = enum_ser(
            &item_enum,
            Ident::new("borsh", Span::call_site()),
        )
        .unwrap();

        insta::assert_snapshot!(pretty_print_syn_str(&actual).unwrap());
    }
    #[test]
    fn borsh_discriminant_true() {
        let item_enum: ItemEnum = syn::parse2(quote! {
            #[borsh(use_discriminant = true)]
            enum X {
                A,
                B = 20,
                C,
                D,
                E = 10,
                F,
            }
        })
        .unwrap();
        let actual = enum_ser(
            &item_enum,
            Ident::new("borsh", Span::call_site())
        )
        .unwrap();

        insta::assert_snapshot!(pretty_print_syn_str(&actual).unwrap());
    }
  ```
  it's solely for the purpose of quick comparison of expansions of two cases.